node_agent/main.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2017 Istio Authors
```

Comment:
The one difference the way Kubernetes is done vs istio Auth is there are a lot of top level directories. I am OK with us having something like this:

cmd\
   istio_ca\
       BUILD
       controller\
       certmanager\
        ca.go
  node_agent\
      BUILD
       na\
       main.go
  proto\
  utils\

This is fine. This is exactly same as what oliver proposed with the difference that instead of calling the directory as ca/ we are calling it cmd/. I think we all have the same idea here.